hudi-spark-datasource/hudi-spark-common/src/test/scala/org/apache/hudi/TestRecordLevelIndexSupport.scala

```diff
@@ -49,7 +49,7 @@ class TestRecordLevelIndexSupport {
 
     // Case 4: EqualTo filters on simple AttributeReference and Literal which should return non-empty result
     testFilter = EqualTo(AttributeReference("_row_key", StringType, nullable = true)(), Literal("row1"))
-    result = RecordLevelIndexSupport.filterQueryWithRecordKey(testFilter, Option.apply(HoodieMetadataField.RECORD_KEY_METADATA_FIELD.getFieldName))
+    result = RecordLevelIndexSupport.filterQueryWithRecordKey(testFilter, Option.apply("_row_key"))
```

Comment:
Can we test both record key field and `RECORD_KEY_METADATA_FIELD` with `RecordLevelIndexSupport.filterQueryWithRecordKey` for all test cases, by parameterizing this test method (including the cases that return empty result)?